### DISCUSSION TITLE
Emotes

### DISCUSSION THREAD

**Discussion by TeamPuzel** [elementary/triage#153] [Q&A] [answered]
Hi!
I think it would be cool to have an emoji button where a popover, like the macOS/iPadOS emote menu would appear. :)
I really need a nice way to access emotes, I don't think there's a menu built in anymore?

I could try write something like this, is there any documentation I could use to figure out how these work? or do I have to just look at existing repositories?

**Comment by danirabbit** [comment, marked as answer]
You can access the emoji picker by right clicking in a text field and selecting "insert emoji". This works with any of the pre-installed apps as well as apps from AppCenter and in the built-in web browser GNOME Web

**Reply by TeamPuzel** [reply]
I'm using a 2015 MacBook Pro for Linux, and gnome web with 2 tabs open makes the fan spin at 100% and the system extremely slow; it's not usable on a dual core system 👀️
And there's a lot of popular apps not present on the AppCenter, though the browser is where the emotes are needed most.
I compiled [https://github.com/cassidyjames/ideogram](url) - it works great but it's difficult to activate

I'm not sure, but since your "insert emoji" button doesn't work in other apps, I assume it's implemented in granite?

I want to activate a menu, like that of Ideogram, from wingpanel. macOS has a feature like that, where a menu bar icon that opens the emoji menu can be enabled from the System Settings. It's not necessary there because the globe key opens the menu on text fields anyway, but I think it would be easier to implement the traditional menu here as a backup for apps not installed from the AppCenter.

Side note, I'm not sure about the discoverability of the existing emoji feature - I had no idea it was a thing, right clicking on text fields is not something I'd ever do

**Reply by TeamPuzel** [reply]
I think it would be enough to just have the current emoji picker pop up from an icon in wingpanel, and automatically insert like Ideogram does. The code for these exists already.

**Reply by TeamPuzel** [reply]
hm by the way the emoji picker seems to lack support for emotes added after 2019

**Comment by TeamPuzel** [comment]
Ok so I think I understand how everything works now, insert emoji just uses the gtk3 .insert_emoji()
the gtk3 implementation is really outdated, I think it would be a good idea to write a custom emoji picker that matches the elementary OS style and actually supports all emotes, is this something that's already being worked on?
